src/nvim/main.c

```diff
@@ -921,6 +921,10 @@ static void remote_request(mparm_T *params, int remote_args, char *server_addr,
       os_errmsg(connect_error);
       os_errmsg("\n");
       os_exit(1);
+    } else if (strequal(server_addr, os_getenv("NVIM"))) {
+      os_errmsg("<ERROR>, maybe you meant --remote instead of --remote-ui.");
```

Comment:
```suggestion
      os_errmsg("Nvim child UI cannot attach to the UI of its parent ($NVIM)");
```

That's good enough for now. Later we can assign a `EXX` number and add a help tag.